### ISSUE TITLE
How can I make a single tree fully overfit the training set?

### ISSUE THREAD

**Issue by miskolc** [dmlc/xgboost#2589]
I have a data set with 800,000 observations and a label for each one of them (true, false). Only 1000 targets belong to the True class, the other ~800,000 belong to the False class.
I'm trying to create a single XGBoost tree to overfit the initial dataset. I'm assuming that a single tree can learn 2^n cases where n is the tree height so normall n=20 would mean a tree with 2^20 > 1,000,000 leafs so enough to overfit my dataset.
The problem is that even with a height of 40 I'm still unable to predict the correct label for each observation in the data set. 

no_epochs=1
xgb_learner_config = {
                'objective'       :'binary:logistic',
                'eta'             : 0.5,
                'max_depth'       : 40, 
                'alpha'           : 0.1,
                'tree_method'     : 'exact'
            }

training_data = xgb.DMatrix(features, label=targets)
xgb_learner = xgb.train(xgb_learner_config, training_data, no_epochs)
prediction_data = xgb.DMatrix(features)
predictions = xgb_learner.predict(prediction_data)
predictions = predictions > 0.5

from sklearn.metrics import precision_recall_fscore_support
precision_recall_fscore_support(targets, predictions, average='binary')
(0.7055555555555556, 0.22597864768683273, 0.3423180592991914, None)

So my F1-score is just 0.34, very slow.
Why is this happening?
What should I do?
What combination of parameters would yield the highest F1-score possible on the training set using a single tree?

**Comment by pommedeterresautee** [comment]
To get more overfitting, you may want to set `lambda` to 0.

**Comment by miskolc** [comment]
@pommedeterresautee  Interesting, I had no idea that Ridge regularization was used by default. Setting lambda to 0 increase my F1 score by more than 0.1 to 0.45  but it's still far from fully overfitting (F1-score = 1.0).

**Comment by Laurae2** [comment]
@miskolc depth 40 is by far not enough, you will need a deeper tree as not every branch is not going to be filled (maybe 1000 depth, but you will exhaust RAM).

See here for an overfitting example with a depth of 30: https://medium.com/data-design/visiting-categorical-features-and-encoding-in-decision-trees-53400fa65931
